pkg/utils/k8sutil.go

```diff
@@ -636,6 +774,14 @@ func EnsureFuncDeployment(client kubernetes.Interface, funcObj *kubelessApi.Func
 				Name:  "FUNC_TIMEOUT",
 				Value: timeout,
 			},
+			v1.EnvVar{
+				Name:  "FUNC_RUNTIME",
+				Value: funcObj.Spec.Runtime,
+			},
+			v1.EnvVar{
+				Name:  "FUNC_MEMORY_LIMIT",
+				Value: funcObj.Spec.Deployment.Spec.Template.Spec.Containers[0].Resources.Limits.Memory().String(),
```

Comment:
I think this should be:
```go
ValueFrom: &v1.EnvVarSource{
    ResourceFieldRef: &v1.ResourceFieldSelector{
        Resource: "limits.memory",
    },
},
```

In particular, this will pick up any other sources of memory limits, like some future auto-sizer thing.

Reply:
this is existing code, I messed up during merges, so i cherry-picked this change.